File: src/tests/fidl/source_compatibility/table-member-add/hlcpp/step_00_before.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/tablememberadd/cpp/fidl.h>  // nogncheck
namespace fidl_test = fidl::test::tablememberadd;

// [START contents]
void use_table(const fidl_test::Profile& profile) {
  if (profile.has_timezone()) {
    printf("timezone: %s", profile.timezone().c_str());
  }
  if (profile.has_temperature_unit()) {
    printf("preferred unit: %s", profile.temperature_unit().c_str());
  }
  for (const auto& entry : profile.UnknownData()) {
    printf("unknown ordinal %lu", entry.first);
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }

File: src/tests/fidl/source_compatibility/table-member-add/go/step_00_before/step_00_before.go
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// +build !build_with_native_toolchain

package main

import (
	lib "fidl/fidl/test/tablememberadd"
	"fmt"
)

// [START contents]
func useTable(profile lib.Profile) {
	if profile.HasTimezone() {
		fmt.Printf("timezone: %s", profile.GetTimezone())
	}
	if profile.HasTemperatureUnit() {
		fmt.Printf("preferred unit: %s", profile.GetTemperatureUnit())
	}
	for ord, data := range profile.GetUnknownData() {
		fmt.Printf("unknown ordinal %d with bytes %v", ord, data.Bytes)
	}
}

// [END contents]

func main() {}
